lib/storage/metadata/mongoclient/MongoClientInterface.js

```diff
@@ -465,6 +465,7 @@ class MongoClientInterface {
     /**
      * In this case we generate a versionId and
      * sequentially create the object THEN update the master.
+     * Master is deleted when version put is a delete marker
```

Comment:
are we already handling the re-creating of Master when a new verison is added after a delete marker? e.g. 

* Create versionned object
* Delete object, i.e. create delete marker → Master version is removed
* Put another version of the object → Master version should be re-created

Reply:
Yes we are, that's what the `upsert: true` param is for, when master is non existent it gets created

Reply:
ok nice, just please make sure the tests do check this